src/lib/integration_test_abstract_engine/ci_interaction.ml

```diff
@@ -0,0 +1,766 @@
+open Core_kernel
+open Async
+open Currency
+open Signature_lib
+open Mina_base
+open Integration_test_lib
+
+(* [alias] is instantiated when command line args are parsed *)
+let alias = ref None
+
+(* [config_path] is instantiated when command line args are parsed *)
+let config_path = ref ""
+
+module Network_config = struct
+  module Cli_inputs = Cli_inputs
+
+  type block_producer_config =
+    { name : string; keypair : Network_keypair.t; libp2p_secret : string }
+  [@@deriving to_yojson]
+
+  type snark_coordinator_config =
+    { name : string; public_key : string; worker_nodes : int }
+  [@@deriving to_yojson]
+
+  type config =
+    { network_id : string
+    ; config_dir : string
+    ; deploy_graphql_ingress : bool
+    ; mina_image : string
+    ; mina_agent_image : string
+    ; mina_bots_image : string
+    ; mina_points_image : string
+    ; mina_archive_image : string
+    ; runtime_config : Yojson.Safe.t
+          [@to_yojson fun j -> `String (Yojson.Safe.to_string j)]
+    ; block_producer_configs : block_producer_config list
+    ; log_precomputed_blocks : bool
+    ; archive_node_count : int
+    ; mina_archive_schema : string
+    ; mina_archive_schema_aux_files : string list
+    ; snark_coordinator_config : snark_coordinator_config option
+    ; snark_worker_fee : string
+    }
+  [@@deriving to_yojson]
+
+  type t =
+    { debug_arg : bool
+    ; genesis_keypairs :
+        (Network_keypair.t Core.String.Map.t
+        [@to_yojson
+          fun map ->
+            `Assoc
+              (Core.Map.fold_right ~init:[]
+                 ~f:(fun ~key:k ~data:v accum ->
+                   (k, Network_keypair.to_yojson v) :: accum )
+                 map )] )
+    ; constants : Test_config.constants
+    ; config : config
+    }
+  [@@deriving to_yojson]
+
+  let testnet_log_filter t = t.config.network_id
+
+  (* TODO: add config file command call here *)
```

Comment:
Are you sure the expand function should call the config file? As far as I'm aware in the GCP case, the expand method is just taking the test_config defined in each of the integration tests and expanding it, e.g. creating key_pairs, constraints and genesis_ledgers. This information is then used in the create and deploy functions, which will need to call commands from the config.